### ISSUE TITLE
[no-unused-expressions] False positive with optional chaining

### ISSUE THREAD

**Issue by ipcjs** [typescript-eslint/typescript-eslint#1138]
**Repro**

```JSON
{
  "rules": {
    "no-unused-expressions": ["error", { "allowShortCircuit": true, "allowTernary": true, "allowTaggedTemplates": true }],
  }
}
```

```TS
class Test {
  test() {
  }
}
declare const t: Test | undefined
t?.test()
```

**Expected Result**

**Actual Result**
```
D:\test\test.ts
  6:1  error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.5.0` |
| `@typescript-eslint/parser`        | `2.5.0` |
| `TypeScript`                       | `3.7.0-dev.20191021` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `v10.15.3` |
| `npm`                              | `6.10.1` |

**Comment by bradzacher** [comment]
Same root cause as #1090 et al

**Comment by foisonocean** [comment]
Is this problem solved?
I'm still facing this problem.

Versions:

```
eslint: v6.7.2
@typescript-eslint/eslint-plugin: v2.12.0
@typescript-eslint/parser: v2.12.0
typescript: v3.7.3
```

**Comment by bradzacher** [comment]
please don't comment on closed issues.
open a new issue and fill in all of the details with a repro.
